nix/haskell.nix

```diff
@@ -87,13 +89,16 @@ let
 
         # cardano-node will want to write logs to a subdirectory of the working directory.
         # We don't `cd $src` because of that.
+				#
+				# TODO: Using the configuration dir works for mainnet, but to add testnet
+        # support, we need to retrieve it properly.
         packages.cardano-wallet-byron.components.benchmarks.restore =
           pkgs.lib.optionalAttrs (!pkgs.stdenv.hostPlatform.isWindows) {
             build-tools = [ pkgs.makeWrapper ];
             postInstall = ''
               wrapProgram $out/bin/restore \
-                --set BYRON_CONFIGS ${pkgs.cardano-node.configs} \
-                --prefix PATH : ${pkgs.cardano-node}/bin
```

Comment:
So — why not use `scripts.mainnet.node` for a pre-configured script?

That script would set `--socket-path` and `database-path` relative to cwd.

We need:
1. The `--database-path` to be in the cache
2. The `--socket-path` to be in a temporary directory to not break CI (#1507)

We cannot simply change the cwd to satisfy both constraints.

Building the launch script ourselves seemed like the safest option.

Reply:
Another option might be to get access to the internal [`mkNodeScript`](https://github.com/input-output-hk/cardano-node/blob/f9d541c51720811897310d09df8b8e43deae9570/nix/scripts.nix#L16-L35) of cardano-node… 🤔

Reply:
Yes fair enough. The config system in cardano-node could be tweaked to better suit our purposes.